src/components/data-pane/field-list.tsx

```diff
@@ -58,6 +71,22 @@ class FieldListBase extends React.PureComponent<FieldListProps, {}> {
       payload: {fieldDef: fieldDef}
     });
   }
+
+  private getFilter(fieldDef: ShelfFieldDef, domain: any[]) {
+    if (typeof fieldDef.field !== 'string') {
+      return;
+    }
+    switch (fieldDef.type) {
+      case ExpandedType.QUANTITATIVE:
+      case ExpandedType.TEMPORAL:
+        return {field: fieldDef.field, range: domain};
+      case ExpandedType.NOMINAL:
+      case ExpandedType.ORDINAL:
+        return {field: fieldDef.field, oneOf: domain};
+      default:
```

Comment:
There should be no default.  Throw error if unsupported type becomes an input here.